schema.json

```diff
@@ -54,23 +54,38 @@
         "default": {},
         "multipleOf": {
             "type": "number",
-            "minimum": 0,
-            "exclusiveMinimum": true
+            "exclusiveMinimum": 0
         },
         "maximum": {
             "type": "number"
         },
-        "exclusiveMaximum": {
-            "type": "boolean",
-            "default": false
-        },
         "minimum": {
             "type": "number"
         },
-        "exclusiveMinimum": {
-            "type": "boolean",
-            "default": false
-        },
+        "oneOf": [
+            {
+                "exclusiveMaximum": {
+                    "type": "maximum"
+                },
+                "exclusiveMinimum": {
+                    "type": "number"
+                }
+            },
+            {
+                "exclusiveMaximum": {
+                    "type": "boolean",
+                    "default": false
+                },
+                "exclusiveMinimum": {
+                    "type": "boolean",
+                    "default": false
+                },
+                "dependencies": {
+                    "exclusiveMaximum": [ "maximum" ],
+                    "exclusiveMinimum": [ "minimum" ]
+                }
+            }
+        ],
         "maxLength": { "$ref": "#/definitions/positiveInteger" },
```

Comment:
@epoberezkin Fixed typo, thanks.

What you suggest in that comment on #185 is essentially what I had in #168 which was rejected.  This was my attempt to split the difference between that and the refusal to use "dependencies" at all that defines @awwright's approach in #185.

In Draft 4 you can only do the boolean form.  We have discussed (but not, IIRC, agreed on) eventually dropping the boolean form and using only numeric forms.

What is your use case for mixing numeric and boolean in the same schema object?  That would be a very confusing schema to me.  Note that if you use the boolean form in the root schema and numeric form in a child schema, or a $ref's schema, this meta-schema allows that just fine.  It's only mixing the approaches in the same object that it avoids.  In this case the `"oneOf"` is specifically to prevent mixing in a single object.